src/extension/ai/manager.ts

```diff
@@ -22,16 +22,28 @@ export class AIManager {
 
     this.client = this.createAIClient()
     this.log = getLogger('AIManager')
-
+    this.log.info('AI: Initializing AI Manager')
     this.completionGenerator = new generate.CompletionGenerator(this.client)
     if (autoComplete) {
       this.registerGhostCellEvents()
     }
   }
 
-  createAIClient(): PromiseClient<typeof AIService> {
+  // N.B. We use arrow notation to ensure this is bound to the AIManager instance.
+  createAIClient = (): PromiseClient<typeof AIService> => {
     const config = vscode.workspace.getConfiguration('runme')
-    const baseURL = config.get<string>('runme.aiBaseURL', 'http://localhost:8877/api')
+    const baseURL = config.get<string>('aiBaseURL', 'http://localhost:8877/api')
+    // TODO(jeremy): The log.info statements prevents the extension from loading at least in debug
+    // mode. My suspicion
+    // is that createAIClient is invoked from the constructor which is invoked at extension
+    // loading time. In logger.ts there is a call to outputChannel = window.createOutputChannel('Runme')
+    // to setup the outputChannel. I wonder if that creates a race condition.
+    // The problem seemed to happen if I call console.log as well. It seemed somewhat intermittent
+    // Which would be consistent with a race condition. I wonder if the problem is specific to
+    // running in debug mode?
+    // this.log.info('AI: Using AI service at: ' + baseURL)
```

Comment:
@sourishkrout Do you have any  idea why these lines would prevent the extension from loading in debug mode in vscode?

Reply:
@jlewi I believe there's no `this` in this context because it's a standalone function based on its definition:

```
  createAIClient = (): PromiseClient<typeof AIService> => {
```

It might result in a static method. It's unclear without doing more research. Did I mean to make this a class member, i.e. what's below?

```
  createAIClient(): PromiseClient<typeof AIService> {
```

Reply:
Also, further up in the constructor, `createAIClient` is being called before the logger is initialized.

Reply:
Of course its something obvious like that. Thanks for the help!